backend/models/bcf.js

```diff
@@ -408,816 +464,639 @@ function getBCFMarkup(issue, account, model, unit) {
-								-parseFloat(_.get(vpXML, "VisualizationInfo.PerspectiveCamera[0].CameraDirection[0].Y[0]._"))
-							];
-							vp.position = [
-								parseFloat(_.get(vpXML, "VisualizationInfo.PerspectiveCamera[0].CameraViewPoint[0].X[0]._")) * scale,
-								parseFloat(_.get(vpXML, "VisualizationInfo.PerspectiveCamera[0].CameraViewPoint[0].Z[0]._")) * scale,
-								-parseFloat(_.get(vpXML, "VisualizationInfo.PerspectiveCamera[0].CameraViewPoint[0].Y[0]._")) * scale
-							];
-
-							vp.fov = parseFloat(_.get(vpXML, "VisualizationInfo.PerspectiveCamera[0].FieldOfView[0]._")) * Math.PI / 180;
-
-							vp.type = "perspective";
-
-						} else if (_.get(vpXML, "VisualizationInfo.OrthogonalCamera[0]")) {
-
-							vp.up = [
-								parseFloat(_.get(vpXML, "VisualizationInfo.OrthogonalCamera[0].CameraUpVector[0].X[0]._")),
-								parseFloat(_.get(vpXML, "VisualizationInfo.OrthogonalCamera[0].CameraUpVector[0].Z[0]._")),
-								-parseFloat(_.get(vpXML, "VisualizationInfo.OrthogonalCamera[0].CameraUpVector[0].Y[0]._"))
-							];
-
-							vp.view_dir = [
-								parseFloat(_.get(vpXML, "VisualizationInfo.OrthogonalCamera[0].CameraDirection[0].X[0]._")),
-								parseFloat(_.get(vpXML, "VisualizationInfo.OrthogonalCamera[0].CameraDirection[0].Z[0]._")),
-								-parseFloat(_.get(vpXML, "VisualizationInfo.OrthogonalCamera[0].CameraDirection[0].Y[0]._"))
-							];
-
-							vp.position = [
-								parseFloat(_.get(vpXML, "VisualizationInfo.OrthogonalCamera[0].CameraViewPoint[0].X[0]._")) * scale,
-								parseFloat(_.get(vpXML, "VisualizationInfo.OrthogonalCamera[0].CameraViewPoint[0].Z[0]._")) * scale,
-								-parseFloat(_.get(vpXML, "VisualizationInfo.OrthogonalCamera[0].CameraViewPoint[0].Y[0]._")) * scale
-							];
-
-							vp.fov = 1.8;
-
-							vp.type = "orthogonal";
-						}
+										const vpComponents = _.get(vpXML, "VisualizationInfo.Components");
 
-						if (_.get(vpXML, "VisualizationInfo.Components")) {
-							const groupDbCol = {
-								account: account,
-								model: model
-							};
+										for (let i = 0; i < vpComponents.length; i++) {
 
-							const vpComponents = _.get(vpXML, "VisualizationInfo.Components");
+											let highlightedGroupObject;
 
-							for (let i = 0; i < vpComponents.length; i++) {
+											// TODO: refactor to reduce duplication?
+											if (vpComponents[i].Selection) {
 
-								let highlightedGroupObject;
+												for (let j = 0; j < vpComponents[i].Selection.length; j++) {
+													for (let k = 0; vpComponents[i].Selection[j].Component && k < vpComponents[i].Selection[j].Component.length; k++) {
+														let objectModel = model;
 
-								// TODO: refactor to reduce duplication?
-								if (vpComponents[i].Selection) {
+														if (settings.federate) {
+															objectModel = ifcToModelMap[vpComponents[i].Selection[j].Component[k]["@"].IfcGuid];
```

Comment:
Also probbaly better to do 
```suggestion
const objectModel = settings.federate ? ifcToModelMap[vpComponents[i].Selection[j].Component[k]["@"].IfcGuid]: model;
```

so we can use const instead of let.